ui/src/app/pages/homepage/homepage.tsx

```diff
@@ -194,37 +213,20 @@ export const Homepage = withUserProfile()(class extends React.Component<
         dataUseAgreementCompleted: (serverConfigStore.getValue().enableDataUseAgreement ?
           (() => getRegistrationTasksMap()['dataUseAgreement'].isComplete(profile))() : true)
       });
-
-      try {
-        const result = await profileApi().syncComplianceTrainingStatus();
-        this.setState({
-          trainingCompleted: getRegistrationTasksMap()['complianceTraining'].isComplete(result)
-        });
-      } catch (ex) {
-        this.setState({trainingCompleted: false});
-        console.error('error fetching moodle training status');
-      }
-
-      try {
-        const result = await profileApi().syncTwoFactorAuthStatus();
-        this.setState({
-          twoFactorAuthCompleted: getRegistrationTasksMap()['twoFactorAuth'].isComplete(result)
-        });
-      } catch (ex) {
-        this.setState({twoFactorAuthCompleted: false});
-        console.error('error fetching two factor auth status');
-      }
-
       this.setState({betaAccessGranted: !!profile.betaAccessBypassTime});
 
       const {workbenchAccessTasks} = queryParamsStore.getValue();
+      const hasRegisteredAccess = hasRegisteredAccessFetch(profile.dataAccessLevel);
+      if (!hasRegisteredAccess || parseInt(workbenchAccessTasks, 2) === 1) {
```

Comment:
nit: this check is inconsistent with the check 3 lines down (L223). `|| workbenchAccessTasks` seems fine to me, I don't think we've specifically defined a contract here - just generally "if any value is specified". If you feel very strongly about `parseInt`, please factor this out so the same logic is applied in both situations.

Reply:
tl;dr I'll use `|| workbenchAccessTasks` to be consistent. When I wrote this I meant to avoid the case where `?workbenchAccessTasks=0` would have the same effect as `?workbenchAccessTasks=1` (since non-empty strings evaluate to `true`). But I think it's reasonable for the empty string to denote `false` and the non-empty string to denote `true` as you imply.